src/app/pages/purchase-list/purchase-list.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, inject, OnInit } from '@angular/core';
+import { ApiService } from '../../services/api.service';
+import { TableModule } from 'primeng/table';
+import { AsyncPipe } from '@angular/common';
+import { TicketOptionPipe } from '../../pipes/ticket-option.pipe';
+import { from, take } from 'rxjs';
+import { IPurchase } from '../../interfaces/purchase.interface';
+
+@Component({
+  selector: 'app-purchase-list',
+  standalone: true,
+  imports: [
+    TableModule,
+    AsyncPipe,
+    TicketOptionPipe
+  ],
+  templateUrl: './purchase-list.component.html',
+  styleUrl: './purchase-list.component.scss'
+})
+export class PurchaseListComponent implements OnInit {
+  private readonly apiService = inject(ApiService);
+  readonly purchaseList$ = this.apiService.purchaseList$;
+  pagination = {
+    pageLimit: 5,
+    currentOffset: 0,
+    totalPages: 1,
+    totalItems: 0,
+  };
+  listLoading = true;
+
+  ngOnInit(): void {
+    this.loadPurchases();
+  }
+
+  setListLoading(loading: boolean): void {
+    this.listLoading = loading;
+  }
+
+  loadNextPage(offset: number): void {
+    this.pagination = {
+      ...this.pagination,
+      currentOffset: offset
+    };
+    this.loadPurchases();
+  }
+
+  loadPurchases(): void {
+    this.setListLoading(true);
+    from(this.apiService.loadPart(this.pagination.currentOffset, this.pagination.pageLimit))
+      .pipe(take(1))
+      .subscribe({
+        next: (response: {
+          items: Array<IPurchase>,
+          totalItems: number,
+        }): void => {
+          this.apiService.setPurchaseList(response.items);
```

Comment:
finde ich etwas unschön, vom apiService die daten zu laden und dann wieder aus der componente die daten zurück in den service zu geben für das pagination.

Das hätte z.B. auch direkt in der `loadPart` passieren können.

Reply:
generell ist der state und pagination logik etwas schwammig zwischen service und dieser komponente. Wenn die `purchaseList$` im service lebt, würde ich auch viele der variablen wie  pagination.totalPages im service sehen.

oder halt alles in dieser componente nur das loadData im Service.